src/en/charms-offline-strategies.md

```diff
@@ -0,0 +1,145 @@
+done out of simplicity.
+
+#### No proxy and the localhost cloud
+
+This "no proxy" idea is often used extensively with the localhost cloud at both
+the (shell) environment level and at the Juju level. The purpose being to
+ensure the client can connect with the local controller and that the local
+machines can inter-communicate, all without going through a proxy.
+
+Hosts to exclude from the proxy include:
+
+ - localhost & 127.0.0.1 (standard 'no proxy' settings)
+ - the address assigned to the regular Ethernet device
+ - the address assigned to the host's LXD bridge (to talk to the controller via
+   its API)
+ - the entire subnet dedicated to the particular LXD installation
+
+## APT proxy
+
+An HTTP/S or FTP proxy may not be configured to accept requests for APT
+packages. An APT proxy is identical to that of an HTTP/S or FTP proxy except
+that it applies specifically to APT package requests.
+
+Some common implementations include:
+
+ - [`APT-cacher`][upstream-apt-cacher]
+ - [`Apt-Cacher NG`][upstream-apt-cacher-ng]
+ - [`squid`][upstream-squid]
+ - [`squid-deb-proxy`][upstream-squid-deb-proxy] (based on squid)
+
+Juju uses options `apt-ftp-proxy`, `apt-http-proxy`, and `apt-https-proxy` to
+set these proxies, depending on the protocol involved (i.e. FTP, HTTP, or
+HTTPS).
+
+## FTP proxy
+
+A standard FTP proxy. Juju uses the `ftp-proxy` option for this type of proxy.
+
+## APT mirror
+
+Instead of proxying client requests to an internet-based repository it is
+possible to maintain the repository internally. That is, you can have a copy or
+*mirror* of a Ubuntu package repository. This option has a large storage
+requirement and the initial setup/download time is considerable. Regular mirror
+synchronization will also be needed.
+
+Here are some popular mirroring solutions:
+
+ - [`apt-mirror`][upstream-apt-mirror]
+ - [`debmirror`][upstream-debmirror]
+ - [`aptly`][upstream-aptly]
+
+Juju uses the `apt-mirror` option for this.
+
+### HTTP/S server
+
+For any sort of mirror, an HTTP/S server (i.e. a web server) will be required
+to respond to the actual client requests. These are the most common ones:
+
+ - [`Apache`][upstream-apache]
+ - [`nginx`][upstream-nginx]
+ - [`lighttpd`][upstream-lighttpd]
+
+## Juju agent mirror
+
+An agent that gets installed onto a Juju machine is proxied through the
+controller. If the latter cannot satisfy the request via its cache it will
+download the agent from the [official agent site][upstream-agents], and then
+pass it on to the machine. It is possible, however, to set up an agent mirror
+so the remote site is not solicited by the controller. Download the latest few
+agents and configure one of the aforementioned web servers accordingly.
+Updates to the mirrored agents will be needed as new versions of Juju itself
+become available. Another approach is to have the client download agents and
+use the `juju sync-agent-binaries` command to push them to a model.
```

Comment:
does this push them to the model? or the controller

Reply:
The CLI help mentions 'model' a multitude of times. I've asked on IRC. Since no response at this time I'm going with what I have.

Reply:
sync-agent-binaries is just a new name for sync-tools, which I think used to work on the controller. But the command is old so the behaviour may have changed. It would be useful to get some clarity though -

Reply:
Yes, one is an alias of the other.
I will add it to my backlog of unanswered questions and update this file whenever it "matures".